File: lib/06.dart
int A06(String contents) {
  List<int> groups = contents.split('\n\n').map((group) {
    return group.replaceAll('\n', '').split('').toSet().length;
  }).toList();

  return groups.reduce((value, element) => value + element);
}

int B06(String contents) {
  List<int> groups = contents.split('\n\n').map((group) {
    List<List<String>> people = group.split('\n').map((person) => person.split('')).toList();
    return people.elementAt(0).where((q) => people.every((person) => person.contains(q))).length;
  }).toList();

  return groups.reduce((value, element) => value + element);
}

File: lib/06.js
module.exports = {
	A06(contents) {
		const groups = contents.split(/\n\n/g).map((group) => {
			const gr = group.replace(/\n/g, '').split('');
			return new Set(gr).size;
		});

		return groups.reduce((acc, curr) => acc + curr);
	},

	B06(contents) {
		const groups = contents.split(/\n\n/g).map((group) => {
			const people = group.split(/\n/g).map((person) => person.split(''));
			return people[0].filter((q) => people.every((person) => person.includes(q))).length;
		});

		return groups.reduce((acc, curr) => acc + curr);
	},
};
